dotCMS/src/main/java/com/dotcms/translate/GoogleTranslationService.java

```diff
@@ -1,40 +1,54 @@
 package com.dotcms.translate;
 
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
 import com.dotcms.rendering.velocity.viewtools.JSONTool;
 import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
 import com.dotcms.repackage.com.google.common.base.Preconditions;
 import com.dotcms.repackage.com.google.common.base.Strings;
+import com.dotcms.security.apps.AppSecrets;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
 import com.dotmarketing.business.ApiProvider;
 import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
 import com.dotmarketing.portlets.languagesmanager.model.Language;
 import com.dotmarketing.util.Config;
 import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
 import com.dotmarketing.util.json.JSONArray;
 import com.dotmarketing.util.json.JSONObject;
 import com.google.common.collect.ImmutableMap;
 
-import java.io.UnsupportedEncodingException;
-import java.net.URLEncoder;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.function.Function;
 import java.util.stream.Collectors;
+
+import com.liferay.util.StringPool;
+import io.vavr.control.Try;
 import org.apache.commons.lang.StringEscapeUtils;
 
+import javax.servlet.http.HttpServletRequest;
+
 public class GoogleTranslationService extends AbstractTranslationService {
 
     private JSONTool jsonTool;
-    public static final String BASE_URL = "https://www.googleapis.com/language/translate/v2";
+    public static final String BASE_URL = "https://translation.googleapis.com/language/translate/v2";
     private String serviceUrl = Config.getStringProperty("GOOGLE_TRANSLATE_SERVICE_BASE_URL",
         BASE_URL);
     private String apiKey;
     private List<ServiceParameter> params;
+    public static final String GOOGLE_TRANSLATE_APP_CONFIG_KEY = "googleTranslate-config";
+    public static final String API_KEY_VAR = "apiKey";
 
     public GoogleTranslationService() {
-        this(Config.getStringProperty("GOOGLE_TRANSLATE_SERVICE_API_KEY", ""), new JSONTool(), new ApiProvider());
+        this(Config.getStringProperty("GOOGLE_TRANSLATE_SERVICE_API_KEY", StringPool.BLANK), new JSONTool(), new ApiProvider());
```

Comment:
Codacy found an issue: [The String literal "GOOGLE_TRANSLATE_SERVICE_API_KEY" appears 5 times in this file; the first occurrence is on line 51](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5384201)